### PR TITLE
Fields: fix loading forever when fields return empty

### PR DESCRIPTION
**Pull request by gtk-grafana**
Introduced by refactoring due to label value routes.

### CODE REVIEW

File: src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx [hunk 1]
```diff
@@ -174,6 +170,8 @@ export class FieldsBreakdownScene extends SceneObjectBase<FieldsBreakdownSceneSt
       loading: logsScene.state.loading,
     };
 
+    console.log('update body, loading?', logsScene.state.loading);
```
Comment (gtk-grafana) [inline src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:173]: remove

File: src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx [hunk 2]
```diff
@@ -104,18 +104,14 @@ export class FieldsBreakdownScene extends SceneObjectBase<FieldsBreakdownSceneSt
   };
 
   private serviceFieldsChanged = (newState: ServiceSceneState, oldState: ServiceSceneState) => {
-    if (JSON.stringify(newState.fields) !== JSON.stringify(oldState.fields)) {
+    if (JSON.stringify(newState.fields) !== JSON.stringify(oldState.fields) || newState.loading !== oldState.loading) {
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:107]: Ohhhh, right. More reasons to move away from this type of checks.

**Review by matyax — APPROVED** [review]

File: src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx (hunk 2 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:107]: Yeah, it's like a useEffect dependency, but there's no eslint rule to let you know when something is referenced but not in the dep array

Comment (matyax) [reply src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:112]: What would happen if the fields are the same, but the loading state changed? I think the "or" will completely override this check.

Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:112]: should be fine, it'll just update the loading state in that case

Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:112]: But yeah, we'll call `getLabelOptions` uncessarily, splitting this out into two functions, for updating loading state, and updating options might be better

**Pull request merged**